### DISCUSSION TITLE
Using macosx-version-min or similar to build for older macOS

### DISCUSSION THREAD

**Discussion by zcream** [Homebrew/discussions#1978] [Getting started]
I am developing on Catalina but a user is getting an error while on 10.12.6 
To debug I have set the min version to Sierra in my Xcode project. This causes a number of warnings for brew related dependencies.

```
ld: warning: object file (/usr/local/Cellar/fmt/7.1.3/lib/libfmt.a(format.cc.o)) was built for newer macOS version (10.15) than being linked (10.12)
ld: warning: object file (/usr/local/lib/libboost_filesystem.a(path.o)) was built for newer macOS version (10.15) than being linked (10.12)
ld: warning: object file (/usr/local/lib/libboost_filesystem.a(operations.o)) was built for newer macOS version (10.15) than being linked (10.12)
ld: warning: object file (/usr/local/lib/libboost_filesystem.a(directory.o)) was built for newer macOS version (10.15) than being linked (10.12)
ld: warning: object file (/usr/local/lib/libboost_filesystem.a(exception.o)) was built for newer macOS version (10.15) than being linked (10.12)
ld: warning: object file (/usr/local/lib/libboost_filesystem.a(utf8_codecvt_facet.o)) was built for newer macOS version (10.15) than being linked (10.12)
```

Can I specify that I need to build for 10.12 during the brew install for all my packages ?

**Comment by carlocab** [comment]
Homebrew formulae are designed to be run only on the machine they were installed on (or, at least, similarly-configured machines), so there is no way to do this, I'm afraid.

**Reply by zcream** [reply]
@ilovezfs posted a solution to using the 10.12SDK using xcodebuild here - https://github.com/Homebrew/homebrew-core/pull/19296#issuecomment-352874177

I would love a better description regarding how and where the SDK_ROOT and MACOSX_DEPLOYMENT_TARGET should be set and if this will be default behaviour for all builds.

**Reply by carlocab** [reply]
Those are just environment variables you can set, but you shouldn't need to once you've set the minimum version of your Xcode project. However, this won't change the fact that you're using a version of Boost built for a newer macOS than the one you're trying to develop for.

To get around this, you'll need to compile Boost yourself so that it targets an older version of macOS--if this is possible. I'm not sure it is, but the Boost documentation may provide hints for this.

**Reply by zcream** [reply]
brew install builds the boost locally. How do I pass these env variables to brew so it builds against an older SDK. I think that's what what was being hinted in that post. Here is a longer description of the problem and solution.

https://stackoverflow.com/questions/41019101/how-can-i-get-boost-to-compile-for-an-earlier-deployment-target-on-osx

**Reply by carlocab** [reply]
If you're referring to `@ilovezfs`' comment, they were talking about setting these variables in the `install` methods of formulae. The SO answer you linked to says that's it's even more involved than just setting environment variables.

Trying to use `brew` to do this will be a lot of work. It'll require modifying the `boost` formula to do what's explained there. If you wish to insist on doing this, it's probably best to [`brew extract`](https://docs.brew.sh/Manpage#extract---version---force-formula-tap) the `boost` formula into your own tap.

It's probably just simpler to compile Boost without using `brew` with the instructions you've linked to above.
